airbyte-integrations/connectors/source-sftp/src/main/java/io/airbyte/integrations/source/sftp/parsers/SftpFileParserFactory.java

```diff
@@ -0,0 +1,19 @@
+package io.airbyte.integrations.source.sftp.parsers;
+
+import io.airbyte.integrations.source.sftp.enums.SupportedFileExtension;
+
+public class SftpFileParserFactory {
+
+    public static SftpFileParser createInstance(SupportedFileExtension fileExtension) {
+        switch (fileExtension) {
```

Comment:
can we somehow re-use the existing instance of the corresponding parser? probably make it singletone?